eventorder_test.go

```diff
@@ -0,0 +1,403 @@
+	)
+	var (
+		m      eventOrderMethods
+		passed bool
+	)
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	// generate the spec we are going to use
+	// this will create a number of jobs for a number of users
+	// and prescribe the sequence of status codes that the webhook will return for each one e.g. 500, 500, 200
+	spec := m.newTestSpec(users, jobsPerUser)
+	spec.responseDelay = responseDelay
+
+	t.Logf("Starting docker services (postgres & transformer)")
+	var (
+		postgresContainer    *destination.PostgresResource
+		transformerContainer *destination.TransformerResource
+		gatewayPort          string
+	)
+	pool, err := dockertest.NewPool("")
+	require.NoError(t, err)
+	containersGroup, _ := errgroup.WithContext(ctx)
+	containersGroup.Go(func() (err error) {
+		postgresContainer, err = destination.SetupPostgres(pool, t)
+		return err
+	})
+	containersGroup.Go(func() (err error) {
+		transformerContainer, err = destination.SetupTransformer(pool, t)
+		return err
+	})
+	require.NoError(t, containersGroup.Wait())
+	t.Logf("Started docker services")
+
+	t.Logf("Starting webhook destination server")
+	webhook := m.newWebhook(t, spec)
+	defer webhook.Server.Close()
+
+	t.Logf("Preparing rudder-server config")
+	writeKey := trand.String(27)
+	workspaceID := trand.String(27)
+	templateCtx := map[string]string{
+		"webhookUrl":  webhook.Server.URL,
+		"writeKey":    writeKey,
+		"workspaceId": workspaceID,
+	}
+	configJsonPath := workspaceConfig.CreateTempFile(t, "testdata/eventOrderTestTemplate.json", templateCtx)
+	t.Setenv("RSERVER_BACKEND_CONFIG_CONFIG_FROM_FILE", "true")
+	t.Setenv("RSERVER_BACKEND_CONFIG_CONFIG_JSONPATH", configJsonPath)
+
+	t.Logf("Starting rudder-server")
+	t.Setenv("DEPLOYMENT_TYPE", string(deployment.DedicatedType))
+
+	t.Setenv("JOBS_DB_PORT", postgresContainer.Port)
+	t.Setenv("JOBS_DB_USER", postgresContainer.User)
+	t.Setenv("JOBS_DB_DB_NAME", postgresContainer.Database)
+	t.Setenv("JOBS_DB_PASSWORD", postgresContainer.Password)
+	t.Setenv("DEST_TRANSFORM_URL", transformerContainer.TransformURL)
+
+	t.Setenv("RSERVER_WAREHOUSE_MODE", "off")
+	t.Setenv("RSERVER_ENABLE_STATS", "false")
+	t.Setenv("RSERVER_JOBS_DB_BACKUP_ENABLED", "false")
+
+	// generatorLoop
+	t.Setenv("RSERVER_ROUTER_JOB_QUERY_BATCH_SIZE", "500")
+	t.Setenv("RSERVER_ROUTER_READ_SLEEP", "10ms")
+	t.Setenv("RSERVER_ROUTER_MIN_RETRY_BACKOFF", "5ms")
+	t.Setenv("RSERVER_ROUTER_MAX_RETRY_BACKOFF", "10ms")
+	t.Setenv("RSERVER_ROUTER_ALLOW_ABORTED_USER_JOBS_COUNT_FOR_PROCESSING", "1")
+
+	// worker
+	t.Setenv("RSERVER_ROUTER_NO_OF_JOBS_PER_CHANNEL", "100")
+	t.Setenv("RSERVER_ROUTER_JOBS_BATCH_TIMEOUT", "100ms")
+
+	// statusInsertLoop
+	t.Setenv("RSERVER_ROUTER_UPDATE_STATUS_BATCH_SIZE", "100")
+	t.Setenv("RSERVER_ROUTER_MAX_STATUS_UPDATE_WAIT", "10ms")
+
+	// hack
+	// t.Setenv("RSERVER_ROUTER_GUARANTEE_USER_EVENT_ORDER", "false")
```

Comment:
[question] is this a leftover or base for future implementation?

Reply:
uncomment it and the test will fail :)